build_tools/circle/linting.sh

```diff
@@ -161,3 +161,14 @@ then
     echo $bad_deprecation_property_order
     exit 1
 fi
+
+# Check for default doctest directives ELLIPSIS and NORMALIZE_WHITESPACE
+
+doctest_directive="$(find . -type f \( -name "*.py" -o -name "*.rst" \) -exec grep -nw -E "# doctest: \+(ELLIPSIS|NORMALIZE_WHITESPACE)" {} \;)"
```

Comment:
wouldn't `git grep -nw -E "# doctest\: \+(ELLIPSIS|NORMALIZE_WHITESPACE)"` be much faster?

Reply:
I didn't know about `git grep`. There is no way to search only in .py and .rst files but I don't think that is going to be a problem. Though it means searching through more files than required.
I didn't trust `grep` to do the file selection as I've had problems, which is why I used `find`. Plus this way `find` and `grep` are working in parallel.
Not a bash expert, I can test locally and see which is faster?

Reply:
I tested and `git grep` seems to be much faster, even though it's going through more files. We also don't have any occurrence of `# doctest +(ELLIPSIS|NORMALIZE_WHITESPACE)` anywhere anyway, so grep going though more files shouldn't be an issue.

Reply:
Your way is way faster!